Fix some errors reported by cppcheck

cdumez opened the pull request:
#### 6138f16f5f19831d761f1a85b350eb020d67f0f4
<pre>
Fix some errors reported by cppcheck
<a href="https://bugs.webkit.org/show_bug.cgi?id=254853">https://bugs.webkit.org/show_bug.cgi?id=254853</a>

Reviewed by Darin Adler.

Fix some errors reported by cppcheck:
```
Source/WTF/wtf/text/SymbolImpl.cpp:41:30: error: Signed integer overflow for expression &apos;1&lt;&lt;31&apos;. [integerOverflow]
    s_nextHashForSymbol |= 1 &lt;&lt; 31;
                             ^
Source/WebCore/editing/ReplaceSelectionCommand.cpp:1750:12: error: Reference to local variable returned. [returnReference]
    return list;
           ^
Source/WebCore/platform/graphics/HEVCUtilities.cpp:441:12: error: Uninitialized struct member: parameters.codec [uninitStructMember]
    return parameters;
           ^
Source/WebCore/testing/Internals.cpp:2988:32: warning: Possible null pointer dereference: window [nullPointer]
    auto frontendWindowProxy = window-&gt;open(*window, *window, url, emptyAtom(), emptyString()).releaseReturnValue();
                               ^
Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:91:12: error: Reference to local variable returned. [returnReference]
    return downloadProxy;
```

* Source/WTF/wtf/text/SymbolImpl.cpp:
(WTF::SymbolImpl::nextHashForSymbol):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::cssValueKeywordID):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::deepestSingleChildList):
(WebCore::ReplaceSelectionCommand::insertAsListItems):
* Source/WebCore/platform/graphics/HEVCUtilities.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::openDummyInspectorFrontend):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:
(WebKit::DownloadProxyMap::createDownloadProxy):
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::createDownloadProxy):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::receivedPolicyDecision):
(WebKit::WebPageProxy::resumeDownload):
(WebKit::WebPageProxy::downloadRequest):
(WebKit::WebPageProxy::contextMenuItemSelected):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::download):
(WebKit::WebProcessPool::resumeDownload):
(WebKit::WebProcessPool::createDownloadProxy):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::createDownloadProxy):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:

Canonical link: <a href="https://commits.webkit.org/262483@main">https://commits.webkit.org/262483@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e30ae7765a5fa1dda364c13ed7bb64ef86feb2f7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1614 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1645 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1702 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2535 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1734 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1593 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1710 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1708 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1518 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1628 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1467 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1462 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2372 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1464 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1448 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1363 "14 flakes 123 failures") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1372 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1472 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1490 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2538 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/1551 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1508 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1350 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/1671 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1424 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1455 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/360 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1571 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/1711 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/475 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/graphics/HEVCUtilities.h:
```diff
@@ -59,7 +59,7 @@ WEBCORE_EXPORT std::optional<HEVCParameters> parseHEVCDecoderConfigurationRecord
 
 struct DoViParameters {
-    enum class Codec { AVC1, AVC3, HEV1, HVC1 } codec;
+    enum class Codec { AVC1, AVC3, HEV1, HVC1 } codec { Codec::HVC1 };
```
Why default to the `3` value instead of the `0` value?

cdumez replied:
I was trying to match the code above in this file:
```
struct HEVCParameters {
    enum class Codec { Hev1, Hvc1 } codec { Codec::Hvc1 };
```

darinadler commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -2985,6 +2985,8 @@ RefPtr<WindowProxy> Internals::openDummyInspectorFrontend(const String& url)
     auto* inspectedPage = contextDocument()->frame()->page();
     auto* localMainFrame = dynamicDowncast<LocalFrame>(inspectedPage->mainFrame());
     auto* window = localMainFrame ? localMainFrame->document()->domWindow() : nullptr;
```
Seems like we should early exit if `localMainFrame` is null rather than using the `?:` operator.

darinadler commented on Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:
```diff
@@ -69,7 +69,7 @@ void DownloadProxyMap::platformDestroy()
 #endif
 
-DownloadProxy& DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, Ref<API::DownloadClient>&& client, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
+Ref<DownloadProxy> DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, Ref<API::DownloadClient>&& client, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
```
I thought the point here was that the download proxy was kept alive by the map and so the caller didn’t need a reference to it. But I suppose that’s too tricky to reason about, and this is better for security anyway.

cdumez replied:
Yes, you are right about the ownership model. I still made the change since we prefer to use smart pointer more nowadays, even when not strictly needed.

cdumez committed:
Fix some errors reported by cppcheck
https://bugs.webkit.org/show_bug.cgi?id=254853

Reviewed by Darin Adler.

Fix some errors reported by cppcheck:
```
Source/WTF/wtf/text/SymbolImpl.cpp:41:30: error: Signed integer overflow for expression '1<<31'. [integerOverflow]
    s_nextHashForSymbol |= 1 << 31;
                             ^
Source/WebCore/editing/ReplaceSelectionCommand.cpp:1750:12: error: Reference to local variable returned. [returnReference]
    return list;
           ^
Source/WebCore/platform/graphics/HEVCUtilities.cpp:441:12: error: Uninitialized struct member: parameters.codec [uninitStructMember]
    return parameters;
           ^
Source/WebCore/testing/Internals.cpp:2988:32: warning: Possible null pointer dereference: window [nullPointer]
    auto frontendWindowProxy = window->open(*window, *window, url, emptyAtom(), emptyString()).releaseReturnValue();
                               ^
Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:91:12: error: Reference to local variable returned. [returnReference]
    return downloadProxy;
```

* Source/WTF/wtf/text/SymbolImpl.cpp:
(WTF::SymbolImpl::nextHashForSymbol):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::cssValueKeywordID):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::deepestSingleChildList):
(WebCore::ReplaceSelectionCommand::insertAsListItems):
* Source/WebCore/platform/graphics/HEVCUtilities.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::openDummyInspectorFrontend):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:
(WebKit::DownloadProxyMap::createDownloadProxy):
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::createDownloadProxy):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::receivedPolicyDecision):
(WebKit::WebPageProxy::resumeDownload):
(WebKit::WebPageProxy::downloadRequest):
(WebKit::WebPageProxy::contextMenuItemSelected):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::download):
(WebKit::WebProcessPool::resumeDownload):
(WebKit::WebProcessPool::createDownloadProxy):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::createDownloadProxy):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:

Canonical link: https://commits.webkit.org/262483@main
